modules/DNN/NeuronLayer.h

```diff
@@ -36,6 +36,7 @@ namespace DirectGraphicalModels {
 			* @details This function updates the neurons' values as \f$ netValues_{(1\times N)} = weights^{\top}_{(C\times N)}\times values_{(1\times C)} + biases_{((1\times N))}\f$
 			* @note This method updates only the nodes' net values
 			*/
+			DllExport void  dotProdVis(const Mat& values, Mat weights); //new method was added, purpose for it is to multiply visible neurons by weights of hidden layer.
```

Comment:
- Please move the declaration of this method up to be above the documentation (comment between /** and */) for the `ditProd()` method.
- What is the difference between old `ditProd()` and new `dotProdVis()` methods?
- Argument `weights `is better to pass not by value, but by constant reference, _i.e._ `const Mat& weights`

Reply:
In the next changes, I have:

- moved the declaration of this method and added more detailed comment on why new dotProd was needed.
- Argument is passed by constant reference now.